feat: Updated financials blog post (#927)
eslint/website: _posts/2022-02-23-paying-contributors-sponsoring-projects.md

## The guiding principle

Before explaining how ESLint is using the money it collects now, it helps to take a step back to understand the guiding principles upon which this approach is based. Each project has different goals for why they are raising money and how it should be spent. For many projects, the goal is to allow the maintainer(s) to work full time on the project. That is a worthy goal provided that the maintainers actually want to work full time on the project. This is a narrow approach to distributing funds in that most or all of the money goes to one or a few people. 

On ESLint, we discovered that we didn't have any team members interested in working on the project full time. So we thought: what if we went in the complete opposite direction? Given that we know people will only work part time, and probably just in their spare time, how could we distribute the funds as widely as possible for the maximum positive impact?

> When we asked ourselves this question, we realized that we were circling around a common theme: anything that benefits the ESLint ecosystem is worth supporting financially. The first and most obvious thing that benefits the ecosystem is having people maintaining ESLint, so we clearly need to make sure people are paid fairly for spending their time working on the project. But what else benefits the ecosystem? Well, ESLint built on top of a bunch of dependencies, and we'd like to support those projects, so what if we donated some of our funds to those projects? There's also a wide range of plugins, parsers, and extensions that help fuel use of ESLint, so what if we started sponsoring some of the projects as well? And we do get a bunch of contributions from outside contributors...what if we paid them?

Reviewer: ```suggestion
When we asked ourselves this question, we realized that we were circling around a common theme: anything that benefits the ESLint ecosystem is worth supporting financially. The first and most obvious thing that benefits the ecosystem is having people maintaining ESLint, so we clearly need to make sure people are paid fairly for spending their time working on the project. But what else benefits the ecosystem? Well, ESLint is built on top of a bunch of dependencies, and we'd like to support those projects, so what if we donated some of our funds to those projects? There's also a wide range of plugins, parsers, and extensions that help fuel use of ESLint, so what if we started sponsoring some of these projects as well? And we also get a bunch of contributions from outside contributors...what if we paid them?
```

So by aligning around the idea that anything that benefits the ESLint ecosystem is worth funding, we were able to find ways to spread the money we collected more widely, and we're excited to share the results.

## How we are spending our money now

In general, we regularly spend money on the following things.

### Paying team members per hour

Each month, all ESLint team members can submit an invoice for the number of hours they worked on the project. Anything they do on the project counts, whether that is writing code, writing documentation, triaging issues, participating in our Discord server, attending meetings on behalf of ESLint, contributing to our upstream dependencies, and so on. All contributions to an open source project are valuable, and we feel like compensating team members for any time they spend on the project is the right thing to do.

Right now, the per-hour rate for Technical Steering Committee (TSC) members and Reviewers is $80/hour; the per-hour rate for Committers is $50/hour.

### The contributor pool

> One of the things we struggled with was how to fairly pay outside contributors for working on ESLint. We didn't think a standard per-hour rate made much sense because 1) people might contribute without knowing to track their hours ahead of time and 2) trusting people you don't know to accurately report their hours spend is a a system that is too easy to exploit. Still, we felt like it wasn't fair for us to pay for team member contributions but not for outside contributions, and so we created the contributor pool.

Reviewer: ```suggestion
One of the things we struggled with was how to fairly pay outside contributors for working on ESLint. We didn't think a standard per-hour rate made much sense because 1) people might contribute without knowing to track their hours ahead of time and 2) trusting people you don't know to accurately report their hours spend is a system that is too easy to exploit. Still, we felt like it wasn't fair for us to pay for team member contributions but not for outside contributions, and so we created the contributor pool.
```

* [Sindre Sorhus](https://sindresorhus.com/) is a prolific open source developer, and ESLint uses several of his modules, including [`chalk`](https://npmjs.com/package/chalk), [`globals`](https://npmjs.com/package/globals), [`import-fresh`](https://npmjs.com/package/import-fresh), [`strip-ansi`](https://npmjs.com/package/strip-ansi), and [`strip-json-comments`](https://npmjs.com/package/strip-json-comments).
* [debug](https://npmjs.com/package/debug) is a small tool ESLint uses to output debugging messages when you use the `--debug` flag. 
* [lint-staged](https://npmjs.com/package/lint-staged) is a simple precommit hook manager that makes it easy to setup linting in your development process.

We think it's important for open source projects that receive a lot of donations to take care of their dependencies. After all, if your project couldn't exist without those dependencies, then they deserve your support.

You can always check to see which projects ESLint is supporting on our [Open Collective page](https://opencollective.com/eslint#category-CONTRIBUTIONS).

### Supporting the community

> Once we started supporting our upstream dependencies, we though, what about the downstream dependencies? There are a lot of projects related to ESLint that make the ESLint ecosystem better. Whether that be a different set of rules, a custom parser, or anything that makes ESLint better to use. We are currently donating $150 per month to these projects:

Reviewer: ```suggestion
Once we started supporting our upstream dependencies, we thought, what about the downstream dependencies? There are a lot of projects related to ESLint that enrich the ESLint ecosystem, whether that be a different set of rules, a custom parser, or anything that makes ESLint better to use. We are currently donating $150 per month to these projects:
```

Figured better to avoid two "better"s. :-)

* [typescript-eslint](https://github.com/typescript-eslint) is the plugin that makes ESLint work with TypeScript.
* [eslint-plugin-import](https://npmjs.com/package/eslint-plugin-import) is a plugin to help you manage your module imports.
* [eslint-plugin-jsx-a11y](https://npmjs.com/package/eslint-plugin-jsx-a11y) is a plugin to help you ensure your JSX code is accessible.

Going forward, we are looking to continue supporting community projects like these. The strength of ESLint is in the ecosystem built by you, and we want to support as much of that ecosystem as possible.

### Big projects

The last way we decided to spend money is on what we playfully called "big projects." These are projects that take a lot of time and effort, and that the team doesn't have the time, interest, or expertise to accomplish on our own. Every project has a backlog of tasks that seem too big to ever accomplish, and we realized that we could start making our way through that list by hiring professionals to do the work.

> The first of our big projects is an ambitious project to redesign the ESLint website. We had wanted to redesign our website for several years, and tried at least three separate times to do it in the spirit of open source, which volunteers and designing by consensus. Each of those times the efforts failed, and so we decided it was time to hire some folks to help us out.

Reviewer 2: ```suggestion
The first of our big projects is an ambitious project to redesign the ESLint website. We had wanted to redesign our website for several years, and tried at least three separate times to do it in the spirit of open source, with volunteers and designing by consensus. Each of those times the efforts failed, and so we decided it was time to hire some folks to help us out.
```

Typo?

Author: Yup, thanks!

> The first of our big projects is an ambitious project to redesign the ESLint website. We had wanted to redesign our website for several years, and tried at least three separate times to do it in the spirit of open source, which volunteers and designing by consensus. Each of those times the efforts failed, and so we decided it was time to hire some folks to help us out.

Reviewer: ```suggestion
The first of our big projects is an ambitious project to redesign the ESLint website. We had wanted to redesign our website for several years, and tried at least three separate times to do it in the spirit of open source, which involves volunteers and designing by consensus. Each of those times the efforts failed, and so we decided it was time to hire some folks to help us out.
```

Author: Looks like I can’t merge this on the GitHub so I’ll have to wait until I get to the computer next.

Reviewer: Maybe it is not needed--I hadn't seen Milos' earlier suggestion to change this to `with volunteers and designing by consensus`, so with that merged, this alternative isn't really necessary I think.

Author: Ah good call. 👍
